### ISSUE TITLE
Using \DeclareRobustCommand

### ISSUE THREAD

**Issue by eg9** [AlphaZTX/fixdif#2]
_I'd not redefine_ `\d` _for the job and use a different name, to begin with._

Anyway, there is a problem with the line
```
\gdef\d{\relax\ifmmode\@dif\else\expandafter\d@accent\fi}
```
It should really be
```
\DeclareRobustCommand\d{\ifmmode\@dif\else\expandafter\d@accent\fi}
```
Why? Because if `\d` is found in a moving argument (a section title, for instance), it will become part of a `\write` and, when TeX will eventually expand the contents, it will be in *no* mode, so the `\else` branch would be followed.

In the particular case, nothing really bad happens, but only by chance. If you try with `\section{$\int f(x)\d x$}` you will see that the table of contents entry has
```
\int f(x)\relax \relax \d x
```
You're lucky here, because the standard `\d` is robust and so nothing awful happens. But if some package loaded before `fixdif` for some reason changes the meaning of `\d`, weird things might happen.

Check that, with the proposed redefinition, the table of contents entry will have
```
\int f(x)\d  x
```
If you choose the strategy I recommend, you should anyway do something like
```
\DeclareRobustCommand\d{\@dif}
```
Also `\newdif` suffers from the same problem: better doing
```
\def\@newdif#1#2{\AtBeginDocument{% <--- Don't forget
  \@ifdefinable#1{\DeclareRobustCommand#1{\mathinner{#2}\mup@tch}}% <--- Don't forget
}}
```
(or with your hand-made test, if you so prefer).

Also check for unprotected endlines.

**Comment by AlphaZTX** [comment]
Sorry for replying that late, I was suffering a backache after hurting my back.

I'll fix the `\d` command in version 1.2 but I'm confused about whether I should fix the `\newdif` command together because there seems no need to make the `#1` of `\newdif` robust. By the way, the `#2` of `\newdif` should contains more than one tokens like `\mathsf{d}`.

**Comment by AlphaZTX** [comment]
Fixed in version 1.2.
